### PR TITLE
feat(laravel): boolean filter

### PR DESCRIPTION
**Pull request by adryyy10**
Adding new boolean filter in order to filter by boolean attributes

### CODE REVIEW

File: src/Laravel/Eloquent/Filter/BooleanFilter.php [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+        $booleanValues = [
+            'true' => true,
+            'false' => false,
+            '1' => true,
+            '0' => false,
+        ];
+
+        if (array_key_exists($values, $booleanValues)) {
+            $values = $booleanValues[$values];
+        }
```
Comment (soyuka) [inline src/Laravel/Eloquent/Filter/BooleanFilter.php:39]: shouldn't we return early and log in case of a wrong value?

**Comment by soyuka** [comment]
There were missing a few things, I've added a test and registered the filter (https://github.com/api-platform/core/pull/6806/files#diff-1f34d8685c9bbcc2037ef47c41cb6dac916a0fe6fc06e2047b9635e35fa13527R423) I'll rebase this later. As you can see adding a functional test really helped to check it's working properly.

**Comment by soyuka** [comment]
needs https://github.com/api-platform/core/pull/6860

**Comment by soyuka** [comment]
thanks @adryyy10!

**Comment by vinceAmstoutz** [comment]
@adryyy10, could you please add a documentation PR for this feature, targeting the `main` branch in this file: https://github.com/api-platform/docs/blob/main/laravel/filters.md? 

Check our [CONTRIBUTING Guide](https://github.com/api-platform/docs/blob/4.0/CONTRIBUTING.md) and do not hesitate if you need help.

**Pull request merged**